lib/spack/docs/build_systems/rocmpackage.rst

```diff
@@ -0,0 +1,113 @@
+.. Copyright 2013-2021 Lawrence Livermore National Security, LLC and other
+   Spack Project Developers. See the top-level COPYRIGHT file for details.
+
+   SPDX-License-Identifier: (Apache-2.0 OR MIT)
+
+.. _rocmpackage:
+
+-----------
+ROCmPackage
+-----------
+
+The ``ROCmPackage`` is not a build system but a helper package. Like ``CudaPackage``,
+it provides standard variants, dependencies, and conflicts to facilitate building
+packages using GPUs though for AMD in this case.
+
+You can find source for this package at
+`<https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/rocm.py>`__.
+
+^^^^^^^^
+Variants
+^^^^^^^^
+
+This package provides the following variants:
+
+* **rocm**
+
+  This variant is used to enable/disable building with ``rocm``.  
+  The default is disabled (or ``False``).
+
+* **amdgpu_target**
+
+  This variant supports the optional specification of the AMD GPU architecture.
+  Valid values are the names of the GPUs (e.g., ``gfx701``), which are maintained
+  in the ``amdgpu_targets`` property.
+
+^^^^^^^^^^^^
+Dependencies
+^^^^^^^^^^^^
+
+This package defines basic ``rocm`` dependencies, including ``llvm`` and ``hip``.
+
```

Comment:
I would add another sentence or two to remind them that if they want to use a pre-installed version of rocm they must add the following to their packages.yaml: hip (prefix: /opt/rocm/hip) , hsa-rocr-dev (prefix: /opt/rocm) , and llvm-amdgpu ( prefix: /opt/rocm/llvm ), and that they are advised to add the internal hip clang compiler to their compilers.yaml file (cc path /opt/rocm/llvm/bin/clang and cxx path /opt/rocm/llvm/bin/clang++ ). These will NOT be automatically found when you do spack external find --not-buildable and spack compiler find .

Reply:
I'm still a bit unsure if registering the amd clang compiler is a great idea. The way I see it it is a device compiler, and we're only registering host compilers, right? I don't have nvcc registered as a compiler either.

We should just try to make sure that packages use amd's clang just to compile device-code, and otherwise use the default compiler for host code. (There's a few packages that currently set CXX to hipcc, which doesn't seem right).

Reply:
I went through the process of debugging a package that used rocm only to find out they linked both against GNU's OpenMP and (the AMD version of) clang's OpenMP, that was a mess.

Reply:
I'm hesitant to add this information -- especially hard-coded paths -- to the documentation (at least at this time).  

I followed those instructions when I was preparing for an E4S hackathon and they caused problems with the hackathon environment. I don't recall off-hand whether it was the `compilers.yaml` change or the `packages.yaml` file that caused my problems.  I **think** it was the latter but I can't state that with 100% certainty.

How about I ensure a reference to the rocm package?

Reply:
That's understandable. 

I think that rocm is packaged with both hipcc (device compiler) and a version of clang (host compiler). I previously ran into an issue when I used a different host compiler which referenced a different version of glibc. That resulted in errors like ../lib/libumpire.so: undefined reference to `std::__throw_out_of_range_fmt(char const*, ...)@@GLIBCXX_3.4.20' , because the rocm stuff was internally using a different version. Referencing /opt/rocm/llvm/bin/clang++ made my error go away. However, there may be other better ways to resolve this issue